rclrs/src/wait.rs

```diff
@@ -107,6 +115,7 @@ impl WaitSet {
     /// [`WaitSet::new`].
     pub fn clear(&mut self) {
         self.subscriptions.clear();
+        self.guard_conditions.clear();
```

Comment:
I think I should iterate over each guard condition here and set it's `in_use_by_wait_set` flag to false?

Reply:
Updated, but now I'm wondering if we need a `Drop` impl for `WaitSet` that also properly "releases" the guard conditions. @nnmm  thoughts?